File: mobile/src/models/sales/DeliveryNote.ts
export interface DeliveryNote {
  id: string;
  tenant_id: string;
  invoice_id: string;
  note: string | null;
  created_by: string;
  created_at: string;
  invoice_number?: string | null;
  customer_name?: string | null;
}

export interface DeliveryNoteCreate {
  invoice_id: string;
  note?: string | null;
}

File: backend/src/models/invoices/delivery_note.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, DateTime, Text, ForeignKey, Index
from sqlalchemy.dialects.postgresql import UUID
from ...config.database_config import Base


class DeliveryNote(Base):
    __tablename__ = "delivery_notes"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    tenant_id = Column(UUID(as_uuid=True), ForeignKey("tenants.id"), nullable=False)
    invoice_id = Column(UUID(as_uuid=True), ForeignKey("invoices.id"), nullable=False)
    note = Column(Text, nullable=True)
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

    __table_args__ = (
        Index("idx_delivery_notes_tenant_id", "tenant_id"),
        Index("idx_delivery_notes_invoice_id", "invoice_id"),
    )
